src/ssmDocument/commands/createDocumentFromTemplate.ts

```diff
@@ -0,0 +1,119 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import * as vscode from 'vscode'
+import * as YAML from 'yaml'
+import { getLogger, Logger } from '../../shared/logger'
+import * as picker from '../../shared/ui/picker'
+import { openAndSaveDocument } from '../util/util'
+
+import { getDocumentTemplate } from 'aws-ssm-document-language-service'
+
+export interface SsmDocumentTemplateQuickPickItem {
+    label: string
+    description: string
+    filename: string
+    language: string
+    docType: string
+}
+
+const SSMDOCUMENT_TEMPLATES: SsmDocumentTemplateQuickPickItem[] = [
+    {
+        label: localize('AWS.ssmDocument.template.command22Json.label', 'JSON Command Document 2.2'),
+        description: localize(
+            'AWS.ssmDocument.template.command22Json.description',
+            'An Example of a command document using schemaVersion 2.2 in JSON'
+        ),
+        filename: 'example22.command.ssm.json',
+        language: 'ssm-json',
+        docType: 'command',
+    },
+    {
+        label: localize('AWS.ssmDocument.template.automationJson.label', 'JSON Automation Document 0.3'),
+        description: localize(
+            'AWS.ssmDocument.template.automationJson.description',
+            'An Example of a automation document using schemaVersion 0.3 in JSON'
+        ),
+        filename: 'example.automation.ssm.json',
+        language: 'ssm-json',
+        docType: 'automation',
+    },
+    {
+        label: localize('AWS.ssmDocument.template.command22Yaml.label', 'YAML Command Document 2.2'),
+        description: localize(
+            'AWS.ssmDocument.template.command22Yaml.description',
+            'An Example of a command document using schemaVersion 2.2 in YAML'
+        ),
+        filename: 'example22.command.ssm.yaml',
+        language: 'ssm-yaml',
+        docType: 'command',
+    },
+    {
+        label: localize('AWS.ssmDocument.template.automationYaml.label', 'YAML Automation Document 0.3'),
+        description: localize(
+            'AWS.ssmDocument.template.automationYaml.description',
+            'An Example of a automation document using schemaVersion 0.3 in YAML'
+        ),
+        filename: 'example.automation.ssm.yaml',
+        language: 'ssm-yaml',
+        docType: 'automation',
+    },
+]
+
+export async function createSsmDocumentFromTemplate(): Promise<void> {
```

Comment:
consider moving the picker related functionality out to a promptXxx function. This function can call promptXxx, and you can test it in a more isolated capacity.

Reply:
Will do.